inc/Plugin.php

```diff
@@ -60,6 +60,32 @@ function ( GP_Translation $translation ) {
 			}
 		);
 
+		add_action(
+			'gp_originals_imported',
+			function ( $project_id, $originals_added, $originals_existing, $originals_obsoleted, $originals_fuzzied, $originals_error ) {
+				$project = ( new ProjectLocator( $project_id ) )->get_project();
+
+				if ( ! $project ) {
+					return;
+				}
+
+				if ( 0 === max( $originals_existing, $originals_obsoleted, $originals_fuzzied, $originals_error ) ) {
```

Comment:
Why did you choose to include ` $originals_error`  in this check? It basically means that nothing was changed.
Isn't `$originals_fuzzied` already covered by the other action since it calls `$translation->set_status()` which internally uses the `save()` method?

I think we'll find out if we have a test for each case. Ideally `$zip_provider->schedule_generation()`  is only triggered once during the whole import.

Reply:
I'm not _that_ familiar with GlotPress, hence my question in https://github.com/wearerequired/traduttore/issues/127#issuecomment-439843796 about this…

Which other action are you referring to that calls `$translation->set_status()`?

> Ideally `$zip_provider->schedule_generation()` is only triggered once during the whole import.

There isn't really much overhead in calling it more than once though.

Otherwise feel free to suggest some other way to do this or to provide some tests.